fds/services/dvc_service.py

```diff
@@ -65,85 +74,103 @@ def __skip_already_added(self, root, dirs) -> None:
-                    "name": "Step Into",
-                    "value": "Step Into"
-                })
-
-            questions = [
-                {
-                    "type": "expand",
-                    "message": f"What would you like to do with {type} {file_or_dir_to_check} of {convert_bytes_to_readable(dir_size)}?",
-                    "name": "selection_choice",
-                    "choices": choices,
-                    "default": "Add to DVC"
-                }
-            ]
-            answers = PyInquirer.prompt(questions)
-            if answers["selection_choice"] == "Add to DVC":
+            answers = DVCService._get_choice(file_or_dir_to_check=file_or_dir_to_check, path_size=path_size, type=type)
+            if answers["selection_choice"] == DvcChoices.ADD_TO_DVC.value:
                 # Dont need to traverse deep
                 [dirs.remove(d) for d in list(dirs)]
                 return file_or_dir_to_check
-            elif answers["selection_choice"] == "Add to Git":
+            elif answers["selection_choice"] == DvcChoices.ADD_TO_GIT.value:
                 # Dont need to traverse deep
                 [dirs.remove(d) for d in list(dirs)]
                 return
-            elif answers["selection_choice"] == "Ignore":
+            elif answers["selection_choice"] == DvcChoices.IGNORE.value:
                 # We should ignore the ./ in beginning when adding to gitignore
                 # Add files to gitignore
                 append_line_to_file(".gitignore", file_or_dir_to_check[file_or_dir_to_check.startswith('./') and 2:])
                 # Dont need to traverse deep
                 [dirs.remove(d) for d in list(dirs)]
                 return
+        return
 
     def __add(self, add_argument: str):
         chosen_files_or_folders = []
+        # Keep track of dirs which are below threshold size, so we dont iterate the files inside these dirs
+        skipped_dirs = []
         # May be add all the folders given in the .gitignore
         folders_to_exclude = ['.git', '.dvc']
         if add_argument == AddCommands.ALL.value:
             path_to_walk = self.repo_path
         else:
             path_to_walk = f"{self.repo_path}/{add_argument}"
+        # if argument is to add a file
+        if os.path.isfile(path_to_walk) and get_size_of_path(path_to_walk) >= MAX_THRESHOLD_SIZE:
+            # Keep the file in chosen list
+            chosen_files_or_folders = [path_to_walk]
         for (root, dirs, files) in os.walk(path_to_walk, topdown=True, followlinks=False):
             # Now skip the un-necessary folders
             [dirs.remove(d) for d in list(dirs) if d in folders_to_exclude]
             # Skip the already added files/folders
             self.__skip_already_added(root, dirs)
             # First check root
-            folder_to_add = self.__get_to_add_to_dvc(root, dirs, "Dir")
+            folder_to_add = self.__get_to_add_to_dvc(root, dirs, "Dir", skipped_dirs)
```

Comment:
`skipped_dirs` looks like it's always empty when being passed as an argument here.
It looks like it should be a return value of `__get_to_add_to_dvc` instead.

Reply:
No, it will be only for the firstime, if there are any skipped_dirs as in https://github.com/DAGsHub/fds/pull/50/files#diff-7e21dd64c52c4cfa97377c8415b8f12e8c06ce9c3fe1313d1422fd32aa047079R117, then it will work on skipping the directory

Reply:
It still looks like it should be a return value of `__get_to_add_to_dvc` - the only place its content is checked (and not just appended to) is in line 169 below `if root in skipped_dirs:`